cmd/example-app/main.go

```diff
@@ -273,6 +268,10 @@ func (a *app) handleCallback(w http.ResponseWriter, r *http.Request) {
 	oauth2Config := a.oauth2Config(nil)
 	switch {
 	case code != "":
+		if state := r.FormValue("state"); state != exampleAppState {
```

Comment:
"state" is only part of a auth code response[0]. it's not part of a refresh event.

I think it's confusing because we're handling both the auth request redirect response (GET) and a frontend post to re-render the page (POST). Maybe we should switch this so GETs and POSTs are what determine the course of action? e.g.

```
switch r.Method {
case "GET":
    // ... handle state and do code exchange
case "POST":
    // ... try to refresh token
}
```

[0] https://tools.ietf.org/html/rfc6749#section-4.1.2